### PR TITLE
Implement peer discovery

### PR DESCRIPTION
**Pull request by bfollington**
Fixes https://github.com/subconsciousnetwork/subconscious/issues/1084

**We need to consider that neighbors won't be indexed retroactively unless we choose to re-index from sphere creation.**

## Changes
- Introduce "Discover" tab
- Index peers-of-peers as `neighbor`s
- Populate discover view using neighbors
- Also list suggestions in "Add User" section (predicting user intent)

 

https://github.com/subconsciousnetwork/subconscious/assets/5009316/021db477-a12a-47b0-8b0d-0a45b6da7e71

### CODE REVIEW

File: xcode/Subconscious/Shared/Components/Common/Profile/Sheets/FollowNewUserFormSheet.swift [hunk 1]
```diff
@@ -273,6 +309,57 @@ struct FollowNewUserFormSheetModel: ModelProtocol {
...
+                state: state,
+                actions: [
+                    .form(.didField(.reset)),
+                    .form(.petnameField(.reset)),
+                    .form(.didField(.setValue(input: neighbor.identity.description))),
+                    .form(.petnameField(.setValue(input: neighbor.name.description)))
+                ],
+                environment: environment
+            )
+        }
```
Comment (bfollington) [inline xcode/Subconscious/Shared/Components/Common/Profile/Sheets/FollowNewUserFormSheet.swift:337]: Pre-populate the form with the best name + DID

File: xcode/Subconscious/Shared/Components/Notebook/NotebookNavigationView.swift [hunk 2]
```diff
@@ -102,6 +102,7 @@ struct NotebookNavigationView: View {
                         }
                     }
                 }
+                .ignoresSafeArea(.keyboard, edges: .bottom)
```
Comment (bfollington) [inline xcode/Subconscious/Shared/Components/Notebook/NotebookNavigationView.swift:105]: An attempt to fix the strange cutoff BG issue we've seen.

File: xcode/Subconscious/Shared/Services/DataService.swift [hunk 3]
```diff
@@ -186,6 +189,35 @@ actor DataService {
                     )
                 }
             }
+            
+            for change in peerChanges {
+                switch change {
+                case let .remove(petname):
+//                    try database.removeAssociate(petname.join(
+//                        petname: petname
+//                    ))
```
Comment (bfollington) [inline xcode/Subconscious/Shared/Services/DataService.swift:198]: TODO: implement

**Review by gordonbrander — CHANGES_REQUESTED** [review]
A few comments and questions. Otherwise LGTM.

File: xcode/Subconscious/Shared/Components/AppTabView.swift
[the remark this answers is not in the document]
Comment (bfollington) [reply xcode/Subconscious/Shared/Components/AppTabView.swift:1]: Sounds good to me, I think there's room for visual improvement on the `DiscoverUserView` component in general but I didn't want to overcook the first pass.

**Review by gordonbrander — APPROVED** [review]
LGTM

**Pull request merged**